main/main.cpp

```diff
@@ -1571,13 +1571,23 @@ Error Main::setup2(Thread::ID p_main_tid_override) {
 				}
 				display_server = DisplayServer::create(i, rendering_driver, window_mode, window_vsync_mode, window_flags, window_size, err);
 				if (err == OK && display_server != nullptr) {
+					if (display_server->get_name() == "headless") {
```

Comment:
Hm it seems like a workaround to handle this after the fact. I'd expect the create function to return an error if the requested driver is not suitable, instead of falling back to headless automatically.

Reply:
So you'd be in favor of removing this loop that tries to fall back to another display server?
('Driver' and 'display server' terms are a bit mixed in this code. There are plans to uncouple both, so I guess this logic is going to be overhauled in any case.)

Reply:
Or you mean that there should be a way to filter out the headless before creating it?